From aa7a98b056ec7c13c60fb5b5cf702f5ac4a2bbc3 Mon Sep 17 00:00:00 2001
From: Greg Kaiser <gkaiser@google.com>
Date: Wed, 1 Aug 2018 13:15:19 -0700
Subject: [PATCH 8/8] cryptfs: Remove Speck support

Remove the Speck encryption support. It was eventually
decided not to allow Speck in Android P, so this code
is no longer needed and wasn't used outside of testing.

Note we don't just "git revert" the original commit
(38723f23ff521e95ed295f500e6529c52a994c9f) because we want
to retain the infrastructure for allowing new types of
crypto algorithms in the future.

Bug: 112009351
Test: Attempted to setup a device with ro.crypto.fde_algorithm set to Speck, and Speck was rejected and the system defaulted to AES.
Change-Id: I69a8b4e8632f8d30b5b54783cb986ab42d4397d9
(cherry picked from commit 17059fe53b7f535886918d06fea58a90930862fc)
---
 cryptfs.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/cryptfs.cpp b/cryptfs.cpp
index 03d2afe..be8e9d5 100644
--- a/cryptfs.cpp
+++ b/cryptfs.cpp
@@ -467,10 +467,6 @@ constexpr CryptoType default_crypto_type = CryptoType()
 
 constexpr CryptoType supported_crypto_types[] = {
     default_crypto_type,
-    CryptoType()
-        .set_property_name("Speck128/128-XTS")
-        .set_crypto_name("speck128-xts-plain64")
-        .set_keysize(32),
     // Add new CryptoTypes here.  Order is not important.
 };
 
-- 
2.17.1

